source/multigrid/mg_transfer_prebuilt.cc

```diff
@@ -85,15 +85,14 @@ MGTransferPrebuilt<VectorType>::prolongate(const unsigned int to_level,
   Assert((to_level >= 1) && (to_level <= prolongation_matrices.size()),
          ExcIndexRange(to_level, 1, prolongation_matrices.size() + 1));
 
-#ifdef DEBUG
-  if (this->mg_constrained_dofs != nullptr)
-    Assert(this->mg_constrained_dofs->get_user_constraint_matrix(to_level - 1)
-               .get_local_lines()
-               .size() == 0,
-           ExcNotImplemented());
-#endif
-
-  prolongation_matrices[to_level - 1]->vmult(dst, src);
+  VectorType copy_src(src);
```

Comment:
would you mind changing this to only make the copy if this the following ``if`` is indeed true? You will likely need an else branch for the vmult() without the vector copy.